controllers/storagecluster/provider_server.go

```diff
@@ -63,6 +67,30 @@ func (o *ocsProviderServer) ensureCreated(r *StorageClusterReconciler, instance
 		return res, nil
 	}
 
+	if os.Getenv(onboardingSecretGeneratorImage) == "" {
+		r.Log.Info("onboardingSecretGeneratorImage env var is not set.")
+	} else {
+		onboardingSecretGeneratorJob := &batchv1.Job{}
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: onboardingJobName}, onboardingSecretGeneratorJob)
+
+		if err != nil && errors.IsAlreadyExists(err) {
+			for _, condition := range onboardingSecretGeneratorJob.Status.Conditions {
+				if condition.Type == batchv1.JobComplete {
```

Comment:
- what happens if the job is completed but it failed to create public key?
- rather than only logging the errors in the job you need to exit w/ a failure (if there's one) like, os.Exit(1)

Reply:
I am not sure i can handle this condition need your help, Taking help over DM